### ISSUE TITLE
clip2tri class design issue.

### ISSUE THREAD

**Issue by Talkless** [raptor/clip2tri#5]
Hi,

It seems that [c2t::clip2tri::triangulate](https://github.com/raptor/clip2tri/blob/5a11e66fef454c33ddfc080c17a9b2b03f7b642b/clip2tri/clip2tri.h#L61) could be marked as static, because it does not use any class member variables.

In fact, there are no member variables in `clip2tri` at all, so maybe `c2t::clip2tri::triangulate()` should be simply become a free function `c2t::triangulate()`, and all private `clip2tri` member functions used in implementation hidden inside `.cpp` file.

It seems that `clip2tri` class existence produces some misunderstanding, as I've seen new'ing `clip2tri` object in Qt source code [qdeclarativepolygonmapitem.cpp line 288](https://code.qt.io/cgit/qt/qtlocation.git/tree/src/imports/location/qdeclarativepolygonmapitem.cpp?h=5.7#n288). There is no object needed as `clip2tri` has no member variables.

**Comment by raptor** [comment]
You are quite right about the class usage.  The whole thing could probably not be a class at all and just be static methods.

I originally designed it as a class like this in case I wanted to expand its functionality in the future.  I am hesitant to change the structure because other people are using it.

**Comment by Talkless** [comment]
Well, at least marking `triangulate()` method as static could work, as guarantee/documentation that no object creation is needed for this particular algorithm, if removing whole class seems too much breaking change.

**Comment by zokai** [comment]
Hi,
I need more documentations for clip2tri.triangulate() function.
1) Do we need to set and pass boundingPolygon?
2) Do holes need to be CCW and the outer loop CW? or it doesn't matter?

Thanks

**Comment by zokai** [comment]
I found the answer myself. 
boundingPolygon is the outer loop polygon and inputPolygons are the holes.

**Comment by papalqi** [comment]
> 我自己找到了答案。
> boundingPolygon是外环多边形，inputPolygons是孔。

thanks bro
